dbms/src/AggregateFunctions/AggregateFunctionStatisticsSimple.h

```diff
@@ -139,8 +136,190 @@ struct VarMomentsDecimal
             throw Exception("Decimal math overflow", ErrorCodes::DECIMAL_OVERFLOW);
         return convertFromDecimal<DataTypeDecimal<T>, DataTypeNumber<Float64>>(tmp / (m0 - 1), scale);
     }
+};
+
+template <typename T, size_t _level>
+struct HighOrderMoments
+{
+    T m[_level + 1]{};
+
+    void add(T x)
+    {
+        ++m[0];
+        m[1] += x;
```

Comment:
And possibly remove struct VarMoments, because this struct can be generalization?